### PR TITLE
Generate docs for dashboard_list

### PR DESCRIPTION
**Pull request by nmuesch**
Generate documentation for dashboard_list resource + data-source
Based on work started in #798

### CODE REVIEW

File: docs/resources/dashboard_list.md [hunk 1]
```diff
@@ -25,67 +27,76 @@ resource "datadog_dashboard_list" "new_list" {
...
+- **name** (String) The name of the Dashboard List
 
-The following arguments are supported:
+### Optional
 
--   `name`: (Required) The name of this Dashboard List.
--   `dash_item`: (Optional) An individual dashboard object to add to this Dashboard List. If present, must contain the following:
-    -   `type`: (Required) The type of this dashboard. Available options are: `custom_timeboard`, `custom_screenboard`, `integration_screenboard`, `integration_timeboard`, and `host_timeboard`
-    -   `dash_id`: (Required) The ID of this dashboard.
+- **dash_item** (Block Set) A set of dashbaord items that belong to this list (see [below for nested schema](#nestedblock--dash_item))
```
Comment (matt-miller-ddog) [inline docs/resources/dashboard_list.md:85]: Typo: `dashbaord`

File: docs/data-sources/dashboard_list.md [hunk 2]
```diff
@@ -1,23 +1,30 @@
...
+## Schema
+
+### Required
+
+- **name** (String) A dashboard list name to limit the search.
+
+### Optional
 
--   `name`: (Required) A dashboard list name to limit the search.
+- **id** (String) The ID of this resource.
```
Comment (matt-miller-ddog) [inline docs/data-sources/dashboard_list.md:28]: I don't see anything using an input `id` field in https://github.com/DataDog/terraform-provider-datadog/blob/master/datadog/data_source_datadog_dashboard_list.go ; how does this ID option work?

File: docs/data-sources/dashboard_list.md [hunk 3]
```diff
@@ -1,23 +1,30 @@
...
```
Comment (matt-miller-ddog) [inline docs/data-sources/dashboard_list.md:29]: Also: dashboard list IDs are ints, not strings.

File: docs/data-sources/dashboard_list.md (hunk 2 above)
Comment (zippolyte) [reply docs/data-sources/dashboard_list.md:28]: `id` is the field where terraform stores the resource ID in its state, that you can also reference in other parts of your terraform config.
It's always stored as a string by terraform itself

File: examples/resources/datadog_dashboard_list/resource.tf [hunk 4]
```diff
@@ -0,0 +1,60 @@
+# Create a new Dashboard List with two Dashboards
```
Comment (zippolyte) [inline examples/resources/datadog_dashboard_list/resource.tf:1]: nit: can we fix the formatting and spacing in there ?

Comment (nmuesch) [reply examples/resources/datadog_dashboard_list/resource.tf:1]: I ran a `terraform fmt` on this file and committed the change.

**Review by zippolyte — APPROVED** [review]

**Review by matt-miller-ddog — APPROVED** [review]

**Pull request merged**